auth/auth.go

```diff
@@ -0,0 +1,26 @@
+package auth
+
+import (
+	"time"
+
+	"golang.org/x/net/context"
+)
+
+// AuthenticationProvider provides helper methods to convert tokens to sessions
+// using our own internal authorization services
+type AuthenticationProvider interface {
```

Comment:
Are we expecting this interface to grow in size? If it'll always just be a RecoverSession method we could just pass around the method directly